plugins/_chatactions.py

```diff
@@ -195,30 +258,112 @@ async def DummyHandler(ult):
-        except Exception as er:
-            LOGS.exception(er)
-    key = udB.get_key("CHATBOT_USERS") or {}
-    if e.text and key.get(e.chat_id) and sender.id in key[e.chat_id]:
-        msg = await get_chatbot_reply(e.message.message)
-        if msg:
-            sleep = udB.get_key("CHATBOT_SLEEP") or 1.5
-            await asyncio.sleep(sleep)
-            await e.reply(msg)
-    chat = await e.get_chat()
-    if e.is_group and sender.username:
-        await uname_stuff(e.sender_id, sender.username, sender.first_name)
-    elif e.is_private and chat.username:
-        await uname_stuff(e.sender_id, chat.username, chat.first_name)
-    if detector and is_profan(e.chat_id) and e.text:
-        x, y = detector(e.text)
-        if y:
-            await e.delete()
+    xxrep = await check_reply_to(e)
+
+    if xxrep:
+        sender = await e.get_sender()
+        if not isinstance(sender, types.User) or sender.bot:
+            return
+        if check_echo(e.chat_id, e.sender_id):
+            try:
+                await e.respond(e.message)
+            except Exception as er:
+                LOGS.exception(er)
+        key = udB.get_key("CHATBOT_USERS") or {}
+        if e.text and key.get(e.chat_id) and sender.id in key[e.chat_id]:
+            # Simulate typing indicator
+            async with e.client.action(e.chat_id, "typing"):
+                msg = await get_chatbot_reply(e.message.message)
+                if msg:
+                    sleep = udB.get_key("CHATBOT_SLEEP") or 1.5
+                    await asyncio.sleep(sleep)
+
+                    # Check if the message length exceeds a certain threshold
+                    if len(msg) > 4096:
+                        # Create a temporary text file
+                        with tempfile.NamedTemporaryFile(
+                            mode="w+", delete=False
+                        ) as temp_file:
+                            temp_file.write(msg)
+
+                        # Send the text file with a caption
+                        await e.client.send_file(
+                            e.chat_id,
+                            temp_file.name,
+                            caption="Here is the response in a text file.",
+                        )
+
+                        # Delete the temporary text file
+                        os.remove(temp_file.name)
+                    else:
+                        # Send the message directly
+                        await e.reply(msg)
+
+        chat = await e.get_chat()
+        if e.is_group and sender.username:
+            await uname_stuff(e.sender_id, sender.username, sender.first_name)
+        elif e.is_private and chat.username:
+            await uname_stuff(e.sender_id, chat.username, chat.first_name)
+        if detector and is_profan(e.chat_id) and e.text:
+            x, y = detector(e.text)
+            if y:
+                await e.delete()
+
+
+@ultroid_bot.on(events.NewMessage(incoming=True))
```

Comment:
what is oracleBot ?

Reply:
OracleBot is a branch of Googles Gemini pro. Or Googles Genrative text. 

Basically it's the biggest chat bot available themed in a sense. 

If you like we can literally called it Ultroid AI and it will introduce itself as Ultroids AI

You can set chatbot bases.

Reply:
If you want me to remove Oracle from the PR i can.. 

then another review would be good :)